### ISSUE TITLE
Add field "title" to L.PathOptions

### ISSUE THREAD

**Issue by danallsop** [ScottLogic/jumpstart-Leaflet#115]
> **Migrated from:** https://github.com/Leaflet/Leaflet/issues/4179
> **Original author:** danallsop
> **Original created:** 2016-01-29T12:02:31Z
---

Hello!
I propose to **add** a field title in the options for `L.PathOptions` to add the field value in the SVG tag `g>path>title`. To display the native tooltips.
![leaflet issue 2](https://cloud.githubusercontent.com/assets/6443173/12675127/2481ad56-c699-11e5-9dba-fb511fe4353d.png)
![leaflet issue](https://cloud.githubusercontent.com/assets/6443173/12675095/ed0ddf84-c698-11e5-88c8-4ed6ad9cc058.png)
Also possible is another option: return the tag object g, the field for manual editing.
Thanks! :)

**Comment by danallsop** [comment]
## Comments (3 total)

---
> **Comment by:** danallsop **on** 2016-01-29T12:17:56Z

I see a problem: what if the line is rendered using a `L.Canvas` rendered instead? This functionality would be specific to the SVG renderer, so I'm not convinced is a 100% good idea :-/

Also, have you tried using the `L.Path.getElement()` method??

---
> **Comment by:** danallsop **on** 2016-02-13T11:05:21Z

Also, is it crossbrowser? Some time ago "title" on SVG was not supported on Webkit, see https://github.com/Leaflet/Leaflet/pull/1331 for a bit of context.

---
> **Comment by:** danallsop **on** 2017-01-25T12:18:07Z

`<title>` support in SVG seems to be across more browsers now. I've verified by adding the `<title>` tag to a `<path>` SVG element of a Leaflet polyline and see it showing in Chrome 55:

![leaflet_polyline_tooltip_mockup](https://cloud.githubusercontent.com/assets/428616/22290156/45dbbc6e-e2f7-11e6-8a5e-a35640ee7699.png)

So might be worth giving this implementation a shot now.

Regarding the argument of lack of support when using `L.Canvas` : Tooltips are an additional hint primarily for non-touch/pointer device interactions. Do we know in which scenarios and device types `L.Canvas` is primarily used? It might still worth adding support for a title option with the caveat of only supporting it on for the SVG renderer.
